wake-lock/idlharness.https.html

```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Wake Lock API IDL tests</title>
+<link rel="author" title="Intel" href="http://www.intel.com">
+<link rel="help" href="https://w3c.github.io/wake-lock/">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/WebIDLParser.js"></script>
+<script src="/resources/idlharness.js"></script>
+<script>
+"use strict";
+
+function doTest([dom_idl, wakelock_idl, wakelock]) {
+  const idl_array = new IdlArray();
+  idl_array.add_untested_idls(dom_idl);
+  idl_array.add_untested_idls('interface Navigator {};');
+  idl_array.add_idls(wakelock_idl);
+  self.wakelock = wakelock;
+  self.request = wakelock.createRequest();
+  idl_array.add_objects({WakeLock: ['wakelock'], WakeLockRequest: ['request']});
+  idl_array.test();
+}
+
+promise_test(() => {
```

Comment:
This isn't actually testing anything, so we should rewrite this as:

```js
promise_test(async () => {
  const idl_array = new IdlArray();
  const domIdl = await fetch("/interfaces/dom.idl").then(r => r.text());
  const wakelockIdl = await fetch("/interfaces/wake-lock.idl").then(r =>
    r.text()
  );
  const wakelock = await navigator.getWakeLock("screen");

  idl_array.add_untested_idls(domIdl);
  idl_array.add_untested_idls("interface Navigator {};");
  idl_array.add_idls(wakelock_idl);

  window.wakelock = wakelock;
  window.request = wakelock.createRequest();

  idl_array.add_objects({
    WakeLock: ["wakelock"],
    WakeLockRequest: ["request"],
  });
  idl_array.test();
}, "Test IDL implementation of WakeLock API");
```

Reply:
Could you clarify why we need to rewrite like this? The test style are consistent with most IDL tests in w-p-t. e.g. https://github.com/w3c/web-platform-tests/blob/master/dom/interfaces.html

Reply:
The "old" style is because there was no universal support for async/await. We now have async/await, so would prefer we use it.

Reply:
Yeah, fixed, it looks more simplified. Thanks!